pages/docs/manual/latest/polymorphic-variant.mdx

```diff
@@ -0,0 +1,273 @@
+---
+title: "Polymorphic Variant"
+description: "The Polymorphic Variant data structure in ReScript"
+canonical: "/docs/manual/latest/polymorphic-variant"
+---
+
+# Polymorphic Variant
+
+Now that we know what [variant types](./variant) are, let's dive into a more specific version, so called polymorphic variants (or poly variants).
+
+First off, here are some key features:
+
+- Poly variants are **structurally typed** (in comparison to **nominally typed** variants). They can be used without an explicit type definition.
+- They allow easier JavaScript interop (compile to strings / objects with predictable `NAME` and `VAL` attribute) and don't need explicit runtime conversions, unlike common variants.
```

Comment:
It would be great if we had some practical use cases/examples:
```rescript
@module("fs")
external openSync: (
  path,
  @bs.string
  [
    | @bs.as("r") #Read
    | @bs.as("r+") #Read_write
    | @bs.as("rs+") #Read_write_sync
    | @bs.as("w") #Write
    | @bs.as("wx") #Write_fail_if_exists
    | @bs.as("w+") #Write_read
    | @bs.as("wx+") #Write_read_fail_if_exists
    | @bs.as("a") #Append
    | @bs.as("ax") #Append_fail_if_exists
    | @bs.as("a+") #Append_read
    | @bs.as("ax+") #Append_read_fail_if_exists
  ],
) => unit = "openSync" 
```
```

Reply:
Maybe as example in the binding importance section below.

Reply:
why do you need `@as` since you can use #\"r+" directly

Reply:
added an example that is more fitting and uses escape syntax